src/popup/components/common/PasswordInput/index.tsx

```diff
@@ -31,26 +31,13 @@ function PasswordInput(
 ) {
   const { children, defaultVisible, ...otherProps } = props;
 
-  const [type, setType] = useState(defaultVisible ? "text" : "password");
-  const [icon, setIcon] = useState(
-    defaultVisible ? IconNames.EYE : IconNames.EYE_SLASH,
-  );
-
-  const toggleVisibility = () => {
-    if (type === "password") {
-      setType("text");
-      setIcon(IconNames.EYE);
-    } else {
-      setType("password");
-      setIcon(IconNames.EYE_SLASH);
-    }
-  };
+  const [visible, setVisible] = useState(defaultVisible);
 
   return (
     <Root>
-      <Input type={type} {...otherProps} />
-      <IconContainer onClick={toggleVisibility}>
-        <Icon name={icon} />
+      <Input type={visible ? "text" : "password"} {...otherProps} />
+      <IconContainer onClick={() => setVisible(!visible)}>
+        <Icon name={visible ? IconNames.EYE : IconNames.EYE_SLASH} />
```

Comment:
```suggestion
        <Icon name={IconNames[visible ? "EYE" : "EYE_SLASH"]} />
```

Since I can't contribute much with logic, here's a pedantic DRY for you 😂

Reply:
This stops type checking from being able to make sure the value is defined. We could later typo `EYE_SLASH` and it won't fail until runtime. So, thanks but no thanks. :stuck_out_tongue: